oracles/twap/src/TWAPTrap.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {Test, console2} from "forge-std/Test.sol";
+
+/**
+ * @title TWAPTrap
+ * @dev A trap example that checks for price deviations for a set of timeseries TWAP oracle data.
+ * This example shows how a trap can be used to detect price manipulation in a TWAP oracle over a large block range.
+ */
+contract TWAPTrap {
+    uint256 public constant THRESHOLD_MULTIPLIER = 10000;
+    uint256 public deviationThreshold; // In basis points (e.g., 500 represents 5%)
+    uint256 public dataPointsToCheck;
+
+    struct PriceDataPoint {
+        uint256 price;
+    }
+
+    /**
+     * @dev Constructor function to initialize the contract.
+     * @param _deviationThreshold The deviation threshold in basis points.
+     * @param _dataPointsToCheck The number of data points to check for price deviations.
+     */
+    constructor(uint256 _deviationThreshold, uint256 _dataPointsToCheck) {
+        deviationThreshold = _deviationThreshold;
+        dataPointsToCheck = _dataPointsToCheck;
+    }
+
+    /**
+     * @dev Collects the latest price data point.
+     * @return The latest price data point.
+     * @dev This function is a placeholder and returns a dummy value.
+     * In a real implementation, it would fetch the latest price data from an oracle.
+     */
+    function collect() external view returns (PriceDataPoint memory) {
+        return PriceDataPoint({price: 0});
```

Comment:
I would prefer to query a mockOracle so we avoid confusion with some of the early examples of traps